src/consensus/pbft/libbyz/replica.cpp

```diff
@@ -2050,37 +2047,35 @@ void Replica::process_new_view(Seqno min, Digest d, Seqno max, Seqno ms)
     {
       encryptor->set_view(prev_view);
     }
+
+    ByzInfo info;
+    bool did_execute = false;
     if (primary() == id())
     {
-      ByzInfo info;
       pc.add_mine(pp);
-      if (execute_tentative(pp, info, pp->get_nonce()))
-      {
-        if (ledger_writer)
-        {
-          last_te_version = ledger_writer->write_pre_prepare(pp, prev_view);
-        }
-      }
+      did_execute = execute_tentative(pp, info, pp->get_nonce());
     }
     else
     {
-      ByzInfo info;
       pc.add_old(pp);
       uint64_t nonce = entropy->random64();
-
-      if (execute_tentative(pp, info, nonce))
-      {
-        if (ledger_writer)
-        {
-          last_te_version = ledger_writer->write_pre_prepare(pp, prev_view);
-        }
-      }
-
+      did_execute = execute_tentative(pp, info, nonce);
       Prepare* p = new Prepare(v, i, d, nonce, nullptr, pp->is_signed());
       pc.add_mine(p);
       send(p, All_replicas);
     }
 
+    if (did_execute && ledger_writer)
```

Comment:
I think this would be clearer with a top level check on did_execute and two checks for ledger_writer and req_in_pp nested.

Reply:
true, thanks